Dockerfile

```diff
@@ -0,0 +1,22 @@
+ARG GO_VERSION=1.18
+ARG ALPINE_VERSION=3.16
+
+### Build binary
+FROM golang:${GO_VERSION}-alpine${ALPINE_VERSION} as build-binary
+ENV GOOS=linux \
+    GOARCH=amd64 \
+    CGO_ENABLED=0
+COPY . /go/src/github.com/deepmap/oapi-codegen
+WORKDIR /go/src/github.com/deepmap/oapi-codegen
+RUN go build -o ./bin/oapi-codegen ./cmd/oapi-codegen/oapi-codegen.go
+
+### User/passwd layer
+FROM alpine:${ALPINE_VERSION} as user-passwd
+RUN addgroup -S oapi-codegen && adduser -S oapi-codegen -G oapi-codegen
+
+### Image
+FROM scratch as image
```

Comment:
Alternatively we could use Chainguard's lightweight undistro images

Reply:
To be honest, i've never used Chainguard's  undistro images... 
I made a change: added possibility to change base image when we want on build.